Stock.AppService/Services/ProviderService.cs

```diff
@@ -0,0 +1,41 @@
+using Stock.AppService.Base;
+using Stock.Model.Entities;
+using Stock.Repository.LiteDb.Interface;
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Stock.AppService.Services
+{
+    public class ProviderService : BaseService<Provider>
+    {
+        public ProviderService(IRepository<Provider> repository) : base(repository)
+        {
+        }
+
+        public new Provider Create(Provider entity)
+        {
+            if (this.NombreUnico(entity.Name))
+            {
+                return base.Create(entity);
+            }
+
+            throw new System.Exception("The name of provider is already in use");
```

Comment:
Si decidís usar excepciones, estaría bueno que vos definas tu excepción, algo así como "ValiditationException" por ejemplo, y no usar la excepción nativa

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/exceptions/creating-and-throwing-exceptions